samples/Store/Domain/Cart.fs

```diff
@@ -38,6 +38,7 @@ module Fold =
         let ofSnapshot (s: Events.Compaction.State) : State =
             { items = [ for i in s.items -> { skuId = i.skuId; quantity = i.quantity; returnsWaived = i.returnsWaived } ] }
     let initial = { items = [] }
+    let originEventType = nameof Events.Snapshotted
```

Comment:
snapshotEventCaseName ?

Reply:
maybe put a comment in there explaining that it might not always be `nameof` ?

Reply:
Is there something we can use from TypeShape or FsCodec? `caseNameOf Event`?

How does FsCodec decide what string to use for the event type? Can we hook into that?

Reply:
Alternatively, this might not be the worst idea

```fsharp
let snapshotEventCaseName = codec.Encode((), Snapshotted).EventType 
```

Reply:
Oh this falls into a "fun with reflection" territory I'm not super comfortable with 😅 

https://github.com/eiriktsarpalis/TypeShape/blob/main/src/TypeShape/Applications/UnionContract.fs#L97-L100

I think we could extract something like this:

```fsharp
let getCaseName case =
        let t = case.GetType()
        let cases = FSharpType.GetUnionCases(t, true)
        let tagIdx = FSharpValue.PreComputeUnionTagReader(t, true) case
        let caseInfo = cases[tagIdx]
        caseInfo.GetCustomAttributes()
        |> Seq.tryPick (function :? DataMemberAttribute as dm -> Some dm.Name | _ -> None)
        |> Option.defaultValue(caseInfo.Name)
```

Reply:
yes, ideally that'd go as a PR upstream

Also in the [docs wouldnt it be nice section](https://github.com/jet/equinox/blob/master/DOCUMENTATION.md#wouldnt-it-be-nice---equinoxeventstoredb) I have:

> Provide a facility in FsCodec.IEventCodec to walk the Event DU to generate a list of event types; use that to generate the server-side event loading filter e.g. when a Decider used a highly selective subset of the known Event Types

i.e. if it can be generalised in some way, that'd be nice

Reply:
(none of this should be a blocker; it's perfectly reasonable to file it as an issue here and/or in TypeShape)

Reply:
> i.e. if it can be generalised in some way, that'd be nice

I think it can be.

```fsharp
let caseName (caseInfo: UnionCaseInfo) =
    caseInfo.GetCustomAttributes()
    |> Seq.tryPick (function :? DataMemberAttribute as dm -> Some dm.Name | _ -> None)
    |> Option.defaultValue(caseInfo.Name)
let getCaseNames<'t> =
    let t = typeof<'t>
    let cases = FSharpType.GetUnionCases(t, true)
    [| for caseInfo in cases -> caseName caseInfo |]
```